snapmodel_policy.go

```diff
@@ -0,0 +1,141 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2019 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package secboot
+
+import (
+	"encoding/base64"
+	"encoding/binary"
+	"errors"
+	"fmt"
+
+	"github.com/canonical/go-tpm2"
+	"github.com/snapcore/snapd/asserts"
+
+	"golang.org/x/xerrors"
+)
+
+const (
+	modelGradeUnset uint8 = iota
+	modelGradeSecured
+	modelGradeSigned
+	modelGradeDangerous
+)
+
+func modelGradeEnum(grade asserts.ModelGrade) (uint8, error) {
```

Comment:
as discussed this should be implemented directly by asserts as ModelGrade.ToByte or similar, it's not a blocker to land this, but I expect the values used to diverge from what is used at the moment here. I will work on that ASAP.

Reply:
Thanks. Also, it doesn't have to be a byte - if you think there's a chance that it might need more than a byte in the future, then any other fixed size is ok

Reply:
yes, maybe is better to consider future extensions/extensibility and go for 32 bits